src/main/java/org/opentripplanner/graph_builder/services/osm/EdgeNamer.java

```diff
@@ -50,25 +51,29 @@ public static EdgeNamer fromConfig(NodeAdapter root, String parameterName) {
       var osmNaming = root
         .of(parameterName)
         .summary("A custom OSM namer to use.")
-        .since(OtpVersion.V2_0)
-        .asString(null);
+        .since(OtpVersion.V1_5)
+        .asEnum(EdgeNamerType.DEFAULT);
       return fromConfig(osmNaming);
     }
 
     /**
      * Create a custom namer if needed, return null if not found / by default.
      */
-    public static EdgeNamer fromConfig(String type) {
+    public static EdgeNamer fromConfig(EdgeNamerType type) {
       if (type == null) {
         return new DefaultNamer();
       }
-
       return switch (type) {
-        case "portland" -> new PortlandCustomNamer();
-        default -> throw new IllegalArgumentException(
-          String.format("Unknown osmNaming type: '%s'", type)
-        );
+        case PORTLAND -> new PortlandCustomNamer();
+        case SIDEWALKS -> new SidewalkNamer();
+        case DEFAULT -> new DefaultNamer();
       };
     }
   }
+
+  enum EdgeNamerType {
+    DEFAULT,
+    PORTLAND,
+    SIDEWALKS,
```

Comment:
Right now it does. Presumably the idea was that you have call the base logic in your custom implementation if you want to "stack" the namers.

I can move the code into a separate graph builder module to make them independent of each other. Do you want that?

I went with the custom namer because it was an already established pattern and kept churn low.

Reply:
I don't expect to be using this myself, and this module is quite self-contained so I don't have a strong preference myself. I was making sure this was a conscious choice and it won't discourage adoption of the sidewalk namer by others. Seems like it would be easy to refactor later if it's ever a problem.